GooglePlayInstant/Editor/QuickDeploy/AccessTokenGetter.cs

```diff
@@ -115,60 +119,104 @@ public static void GetAuthCode(Action<AuthorizationCode> onAuthorizationCodeActi
+                {"grant_type", OAuth2RefreshTokenGrantType}
+            };
+            RequestToken(grantDictionary, postTokenAction);
+        }
+        
+        /// <summary>
+        /// Sends an HTTP request to OAuth2 token uri to retrieve, process and store needed tokens.
+        /// </summary>
+        /// <param name="grantDictionary">A dictionary containing OAuth2 grant type and grant values to be used when
+        /// requesting access token.</param>
+        /// <param name="postTokenAction"></param>
+        private static void RequestToken(Dictionary<string, string> grantDictionary, Action postTokenAction)
+        {
+            var credentials = OAuth2Credentials.GetCredentials();
+            var formData = grantDictionary
+                .Union(new Dictionary<string, string>
+                {
+                    {"client_id", credentials.client_id},
+                    {"client_secret", credentials.client_secret}
+                }).ToDictionary(pair => pair.Key, pair => pair.Value);
             WwwRequestInProgress.TrackProgress(
-                HttpRequestHelper.SendHttpPostRequest(credentials.token_uri, formData, null),
-                "Requesting access token",
+                HttpRequestHelper.SendHttpPostRequest(credentials.token_uri, formData, null), "Requesting access token",
                 completeTokenRequest =>
                 {
-                    var responseText = completeTokenRequest.text;
-                    var token = JsonUtility.FromJson<GcpAccessToken>(responseText);
-                    if (string.IsNullOrEmpty(token.access_token))
-                    {
-                        throw new Exception(string.Format(
-                            "Attempted to request access token and received response with error {0} and text {1}",
-                            responseText, completeTokenRequest.error));
-                    }
-
-                    AccessToken = token;
+                    HandleTokenResponse(completeTokenRequest);
                     postTokenAction();
                 });
         }
 
+        /// <summary>
+        /// Handles received response from a token request by parsing the response to update access and refresh token
+        /// values if available, as well as throwing an error when there was a failure getting required tokens.
+        /// </summary>
+        /// <param name="completeTokenRequest">A www instance holding a token  request that has received a response.</param>
+        private static void HandleTokenResponse(WWW completeTokenRequest)
+        {
+            var responseError = completeTokenRequest.error;
+            var responseText = completeTokenRequest.text;
+            if (!string.IsNullOrEmpty(completeTokenRequest.error))
+            {
+                throw new Exception(string.Format(
+                    "Sent request to get access token and received response with error {0}, and text{1}", responseError,
```

Comment:
Missing a space after "text". While we're changing things though, it's preferable to quote these strings (easier to tell if one is empty):
`"Received access token response with error \"{0}\" and text \"{1}\""`